lib/adagrams.rb

```diff
@@ -0,0 +1,120 @@
+require 'csv'
+require 'rake/testtask'
+#WAVE 1
+def draw_letters
+  letters = ('A'..'Z').to_a
+  tally = [9,2,2,4,12,2,3,2,9,1,1,4,2,6,8,2,1,6,4,6,4,2,2,1,2,1]
+  zipped = Hash[letters.map.zip(tally)]
+
+  available_letters = zipped.each_with_object([]) do |(letter, amount), array|
+    array << (letter * amount).split("")
+  end
+
+  available_letters.flatten! #array that holds each letter times the paired value
```

Comment:
you can do this with a little less effort and a little more legibly by putting all the letters in a hash like so...

```ruby
hash = { 
'a':9,
'b':2,
'c':2,
#... etc
}
```

And then pushing the letter into an array the number of times indicated by the hash. That way, if your boss asks you to put in 8 m's instead of whatever is in there now, it's easier to change!